features/FxA-49-cdn/README.md

```diff
@@ -0,0 +1,91 @@
+to overall application start time. By pushing static resources
+closer to the user, the hope is users experience a faster
+percieved start time.
+
+All static resources will be served from <XXX>.
+
+### Measuring Success
+
+* `navigationTiming` data - for DOMContentLoaded
+* `resourceTiming` data of individual resources.
+  The largest of the static assets is `main.js`
+* `loaded` event in metrics - when the user sees the
+  first screen - the user based metric that matters.
+
+It is expected that both `DOMContentLoaded` from navigationTiming
+data and the `loaded` message in internal metrics will decrease,
+especially for users located in geographically distant regions
+from our primary data centers.
+
+### CDN choice
+
+[Amazon CloudFront](https://aws.amazon.com/cloudfront/) will be the
+CDN used to deliver static resources.
+
+### Preventing resource manipulation by 3rd parties
+
+The [Subresource Integrity](https://developer.mozilla.org/docs/Web/Security/Subresource_Integrity) `integrity` attribute will be added to all `script` and `link` tags to ensure static JavaScript and CSS is not manipulated by 3rd parties,
+
+From MDN:
+
+> Subresource Integrity (SRI) is a security feature that enables browsers to
+verify that files they fetch (for example, from a CDN) are delivered without
+unexpected manipulation. It works by allowing you to provide a cryptographic
+hash that a fetched file must match.
+
+#### Subresource Integrity SHA generation
+
+A unique SHA must be generated for each static JavaScript and CSS resource.
+The SHAs are then embedded into HTML as the `integrity` attribute of a `script`
+or `link` tag.
+
+
+This can be done using `grunt` in the build step. Since unique JavaScript
+and CSS bundles are created for each locale, this must be done
+post-l10n processing.
+
+### Updating CDN resources
+
+From the [CloudFront documentation](https://aws.amazon.com/cloudfront/details/):
+
+> When a viewer requests a web page or content using that domain name, Amazon CloudFront determines the best edge location to serve your content. If an edge location doesn’t have a copy of the file requested by the viewer, Amazon CloudFront will pull a copy from the origin server and hold it at the edge location so it’s available for future requests.
+
+Once the initial setup is complete, all user requested resources that
+are not yet on the CDN will be pulled from our servers automatically.
+No additional work is needed on deployment to seed the CDN.
+
+## Work Breakdown
+
+* [ ] Generate subresource integrity SHAs for each static JS and CSS resource.
+  * [ ] Add the `integrity` attribute to all `script` and `link` tags.
+* [ ] Prepend CDN domain name to all static resource URLs embedded in HTML and
+    CSS when run in production mode.
+  * [ ] Domin name should be pulled from configuration.
+* [ ] Decide the domain name to use for the CDN.
+  * [ ] Purchase any necessary SSL certs.
+* [ ] Set up a CloudFront CDN account and initial setup.
```

Comment:
It may be worth adding explicit tasks here for measuring success, especially if you're planning to do an manual before/after test rather than using an ongoing performance graph.  Something like "establish baseline performance numbers" and "measure improvement in performance numbers".